lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php

```diff
@@ -655,6 +655,18 @@ public function resolveVersionAlias($alias)
             case 'latest':
                 return $this->getLatestVersion();
             default:
+                if (substr($alias, 0, 7) == 'current') {
+                    $symbol = substr($alias, 7, 1);
+                    $number = (int) substr($alias, 8);
+
+                    if ($number <= 0) {
+                        return null;
+                    }
+
+                    if ($symbol == "+" || $symbol == "-") {
+                        return $this->getRelativeVersion($this->getCurrentVersion(), (int)($symbol.$number));
+                    }
+                }
```

Comment:
This logic should be abstracted in a function so that it's easily testable and that the switch stays readable.

You also need to add test before merge.